keep/step/step.py

```diff
@@ -167,28 +172,34 @@ def _run_single(self):
             result = self._run_single_async()
```

Comment:
the async should be fixed too (or at least adding a test for that)

Reply:
@shahargl you mean handling it for retry?